saleor/wishlist/models.py

```diff
@@ -0,0 +1,75 @@
+import uuid
+
+from django.db import models
+
+from ..account.models import User
+from ..product.models import Product, ProductVariant
+
+
+class WishlistQuerySet(models.QuerySet):
+    def get_or_create_wishlist_for_user(self, user):
+        try:
+            return user.wishlist
+        except self.model.DoesNotExist:
+            return self.create(user=user)
+
+
+class Wishlist(models.Model):
+    token = models.UUIDField(default=uuid.uuid4, unique=True, editable=False)
+    user = models.OneToOneField(
+        User, related_name="wishlist", on_delete=models.CASCADE, blank=True, null=True
+    )
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    objects = WishlistQuerySet.as_manager()
+
+    def set_user(self, user):
+        self.user = user
+        self.save()
+
+    def get_all_variants(self):
+        return ProductVariant.objects.filter(
+            wishlist_items__wishlist_id=self.pk
+        ).distinct()
+
+    def add_variant(self, variant: ProductVariant):
+        item, is_created = self.items.get_or_create(product_id=variant.product_id)
```

Comment:
Good catch, it will. Though `add_variant` is resistant to these cases as `get_or_create` ensures we don't create a duplicate `wishlist, product` instance and `item.variants.add` already handles duplicates (it doesn't make them).